server/src/resolvers/mapping.ts

```diff
@@ -0,0 +1,147 @@
+import { FieldResolver } from 'nexus'
+import {
+  AttributeCreateWithoutResourceInput,
+  InputCreateWithoutAttributeInput,
+  ColumnCreateWithoutInputsInput,
+  JoinCreateWithoutColumnInput,
+  FindManyResourceIncludeArgs,
+  Photon,
+} from '@prisma/photon'
+
+import {
+  JoinWithColumn,
+  ColumnWithJoins,
+  InputWithColumn,
+  AttributeWithChildren,
+} from 'types'
+
+const clean = (entry: any): any => {
+  delete entry.id
+  delete entry.updatedAt
+  delete entry.createdAt
+  return entry
+}
+
+const buildJoinsFromMapping = (
+  joins: JoinWithColumn[],
+): JoinCreateWithoutColumnInput[] | null =>
+  joins.map(j => {
+    let join: JoinCreateWithoutColumnInput = clean(j)
+    if (j.tables && j.tables.length) {
+      join.tables = { create: j.tables.map(clean) }
+    }
+    return join
+  })
+
+const buildColumnFromMapping = (
+  c: ColumnWithJoins,
+): ColumnCreateWithoutInputsInput | null => {
+  let column: ColumnCreateWithoutInputsInput = clean(c)
+  if (c.joins && c.joins.length) {
+    column.joins = { create: buildJoinsFromMapping(c.joins) }
+  } else {
+    delete column.joins
```

Comment:
Is it necessary to delete this?
And why don't you do it line 32?

Reply:
it is necessary otherwise photon complains if I put null or undefined instead of deleting the field.
The case where we have a join with empty tables (line 32) should never happen but i'll add it anyway, you're right.